src/core/query.js

```diff
@@ -800,7 +801,7 @@ export class Query {
 
     // Remove fields
     if (Array.isArray(json.fields) && json.fields.length > 0) {
-      const fields = Array.concat([], json.fields, PROTECTED_FIELDS);
+      const fields = [].concat(json.fields, PROTECTED_FIELDS);
```

Comment:
This will cause a conflict once #258 gets merged. I haven't merged 258 cause of Tejas' request for changes. I keep forgetting to bring this up.